File: clients/php/Model/Order.php
<?php

namespace Model {
    require_once 'Model/UnitOrder.php';
    require_once 'Stream.php';

    /**
     * Player's (team's) orders
     */
    class Order
    {
        /**
         * Orders for each of your units
         */
        public array $unitOrders;
    
        function __construct(array $unitOrders)
        {
            $this->unitOrders = $unitOrders;
        }
    
        /**
         * Read Order from input stream
         */
        public static function readFrom(\InputStream $stream): Order
        {
            $unitOrders = [];
            $unitOrdersSize = $stream->readInt32();
            for ($unitOrdersIndex = 0; $unitOrdersIndex < $unitOrdersSize; $unitOrdersIndex++) {
                $unitOrdersKey = $stream->readInt32();
                $unitOrdersValue = \Model\UnitOrder::readFrom($stream);
                $unitOrders[$unitOrdersKey] = $unitOrdersValue;
            }
            return new Order($unitOrders);
        }
        
        /**
         * Write Order to output stream
         */
        public function writeTo(\OutputStream $stream): void
        {
            $stream->writeInt32(count($this->unitOrders));
            foreach ($this->unitOrders as $key => $value) {
                $stream->writeInt32($key);
                $value->writeTo($stream);
            }
        }
    }
}

File: clients/go/model/order.go
package model

import "fmt"
import "io"
import . "ai_cup_22/stream"

// Player's (team's) orders
type Order struct {
    // Orders for each of your units
    UnitOrders map[int32]UnitOrder
}

func NewOrder(unitOrders map[int32]UnitOrder) Order {
    return Order {
        UnitOrders: unitOrders,
    }
}

// Read Order from reader
func ReadOrder(reader io.Reader) Order {
    var unitOrders map[int32]UnitOrder
    unitOrdersSize := ReadInt32(reader)
    unitOrders = make(map[int32]UnitOrder)
    for unitOrdersIndex := int32(0); unitOrdersIndex < unitOrdersSize; unitOrdersIndex++ {
        var unitOrdersKey int32
        unitOrdersKey = ReadInt32(reader)
        var unitOrdersValue UnitOrder
        unitOrdersValue = ReadUnitOrder(reader)
        unitOrders[unitOrdersKey] = unitOrdersValue
    }
    return Order {
        UnitOrders: unitOrders,
    }
}

// Write Order to writer
func (order Order) Write(writer io.Writer) {
    unitOrders := order.UnitOrders
    WriteInt32(writer, int32(len(unitOrders)))
    for unitOrdersKey, unitOrdersValue := range unitOrders {
        WriteInt32(writer, unitOrdersKey)
        unitOrdersValue.Write(writer)
    }
}

// Get string representation of Order
func (order Order) String() string {
    stringResult := "{ "
    stringResult += "UnitOrders: "
    unitOrders := order.UnitOrders
    stringResult += "map[ "
    unitOrdersIndex := 0
    for unitOrdersKey, unitOrdersValue := range unitOrders {
        if unitOrdersIndex != 0 {
            stringResult += ", "
        }
        stringResult += fmt.Sprint(unitOrdersKey)
        stringResult += ": "
        stringResult += unitOrdersValue.String()
        unitOrdersIndex++
    }
    stringResult += " ]"
    stringResult += " }"
    return stringResult
}